File: April/Abiji-2020/Golang /Make The String Great/Make The String Great.go
func makeGood(s string) string {
sb := []rune(s)
	stack := make([]byte, 0)

	if len(sb) > 1 {

		for i, _ := range s {

			n := len(stack) 

			if n > 0 && ((s[i]-stack[n-1] == 32) || (stack[n-1]-s[i] == 32)) {

				stack = stack[:n-1]

			} else {
				stack = append(stack, s[i])
			}
		}
		return string(stack)

	} else {
		return s
	}

}

File: April/Abiji-2020/C++/Make The String Great/Make The String Great.cpp
class Solution {
public:
    string makeGood(string s) {
        stack<char> st;
        for (char c:s){
            if (!st.empty() && abs(c-st.top())==32){
                st.pop();
            }
            else{
                st.push(c);
            }
        }
        
        string ans="";
        while(!st.empty()){
            ans =st.top()+ans;
            st.pop();
        }
        return ans;
    }
};
